cmd/editorconfig-checker/main_test.go

```diff
@@ -133,6 +135,68 @@ func TestMainDryRun(t *testing.T) {
 	}
 }
 
+// helper to set a map of environment variables and return a function to use in t.Cleanup to reset once the test completed
+// thanks to https://dev.to/arxeiss/auto-reset-environment-variables-when-testing-in-go-5ec
+func envSetter(envs map[string]string) (closer func()) {
+	originalEnvs := map[string]string{}
+
+	for name, value := range envs {
+		if originalValue, ok := os.LookupEnv(name); ok {
+			originalEnvs[name] = originalValue
+		}
+		_ = os.Setenv(name, value)
+	}
+
+	return func() {
+		for name := range envs {
+			origValue, has := originalEnvs[name]
+			if has {
+				_ = os.Setenv(name, origValue)
+			} else {
+				_ = os.Unsetenv(name)
+			}
+		}
+	}
+}
+
+func TestMainColorSupport(t *testing.T) {
+	type env map[string]string
+	type args []string
+
+	tests := []struct {
+		name string
+		env  env
+		args args
+	}{
+		{"no-envvar-no-arg", env{}, args{}},
+		{"envvar-no-arg", env{"NO_COLOR": "1"}, args{}},
+		{"no-envvar-color-off", env{}, args{"--no-color"}},
+		{"no-envvar-color-on", env{}, args{"--color"}},
+		{"envvar-color-off", env{"NO_COLOR": "1"}, args{"--no-color"}},
+		{"envvar-color-on", env{"NO_COLOR": "1"}, args{"--color"}},
+		{"no-envvar-color-offon", env{}, args{"--no-color", "--color"}},
+		{"no-envvar-color-onoffon", env{}, args{"--color", "--no-color", "--color"}},
+		{"envvar-true", env{"NO_COLOR": "true"}, args{}},
```

Comment:
add a test about NO_COLOR: "yes" and another to "no", you will validate they fall back to true

Reply:
done